### PR TITLE
Integration + Product + Category

### LINKED ISSUE #1: Add real api connect
I see Promise for resolve request and return js object.
Do you want to add a service to connect to a real gql API?

### LINKED ISSUE #2: Fetch single/multiple products based on uuid/sku/catId/query

### LINKED ISSUE #3: Fetch product variants

### LINKED ISSUE #4: Fetch single category

### LINKED ISSUE #5: Fetch category tree

### LINKED ISSUE #16: Connect to Vendure shop-api

### PR DESCRIPTION
**Pull request by Baroshem**
Closes #16 
Closes #1 
Closes #3 
Closes #4 
Closes #5 
Closes #2 

I have created a connection between local Vendure server created with a command `npx @vendure/create my-app` and developed basic getCategory functionality and full getProduct functionality. This is Work in Progress Pull Request because I have to finish getCategory functionality, write unit tests and add docs. I also updated the repo to the latest boilerplate state (that's why there is so many changes).

Product Page
image

Product details
image

### CODE REVIEW

**Comment by atlantis-vsf** [comment]
```
Error: This repo is not allowlisted for Atlantis.
```

File: packages/api-client/src/api/getProduct/defaultQuery.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      variants {
+        id
+        sku
+        name
+        price
+        priceWithTax
+        currencyCode
+      }
+      featuredAsset {
+        source
```
Comment (michaelbromley) [inline packages/api-client/src/api/getProduct/defaultQuery.ts:17]: use the `preview` property any time you get an Asset - the `source` is the source binary and might be very large, whereas `preview` is constrained to a sensible file size.

File: packages/api-client/src/api/getCategory/defaultQuery.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          id
+          name
+          slug
+        }
+        description
+        position
+        slug
+        featuredAsset {
+          id
+          name
```
Comment (michaelbromley) [inline packages/api-client/src/api/getCategory/defaultQuery.ts:23]: If you need to actually display the collection image, get the `preview` property.

File: packages/api-client/src/api/getCategory/defaultQuery.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        }
+        children {
+          id
+          name
+        }
+        filters {
+          code
+        }
+        translations {
```
Comment (michaelbromley) [inline packages/api-client/src/api/getCategory/defaultQuery.ts:40]: Normally you don't need to fetch the `translations` field, as the correct translated values (name, slug, description) are automatically populated into the Collection type at runtime by Vendure, based on the current context (i.e. the current channel's default language, or the current language as specified by the [languageCode query param](https://www.vendure.io/docs/storefront/shop-api-guide/#universal-parameters).

File: packages/api-client/src/api/getCategory/defaultQuery.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          id
+          createdAt
+          updatedAt
+          languageCode
+          name
+          slug
+          description
+        }
+        productVariants {
```
Comment (michaelbromley) [inline packages/api-client/src/api/getCategory/defaultQuery.ts:49]: If the intention is to list all the variants in a collection, I recommend using the `search` query with the corresponding `collectionId` or `collectionSlug` input argument. This is a _much_ more efficient way to get that data.

**Review by michaelbromley — COMMENTED** [review]
Just left a few comments about the usage of the API. Happy to answer any Qs via Slack to help get this project done!

**Comment by Baroshem** [comment]
Add a new getter to get dynamic header navigation fetched from Vendure
image

**Comment by Baroshem** [comment]
Currently added category page. The only thing left is to implement remaining getters. We have a category page listing products that also have correct links to product page. Unfortunately I faced a problem with displaying product prices on category page when using search query with groupByProduct option. It does not return any price. I asked Michael for the possible sollutions to that problem.
image

**Comment by Baroshem** [comment]
Displaying sorting options and filters
image
image

**Comment by Baroshem** [comment]
Also, fixed problem with displaying prices in search query thanks to @michaelbromley 
image

**Comment by Baroshem** [comment]
Category page is now working correctly.
image

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
+name: Run tests
+
+on:
+  push:
+    branches:
+      - main
+      - develop
```
Comment (Mateuszpietrusinski) [inline .github/workflows/test.yml:7]: AFAIK you have only main branch so this is unnecessary

File: packages/api-client/src/api/getProduct/index.ts [hunk 6]
```diff
@@ -1,10 +1,36 @@
...
+import ApolloClient, { ApolloQueryResult } from 'apollo-client';
+import { CustomQuery, Context } from '@vue-storefront/core';
+import { ProductData } from '../../types';
+
+interface ProductParams {
+  id: string;
+  slug: string;
 }
 
+const getProduct = async (context: Context, params: ProductParams, customQuery?: CustomQuery): Promise<ApolloQueryResult<ProductData>> => {
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/getProduct/index.ts:12]: IMO better returning type will be ```Promise<ProductData>```

File: packages/api-client/src/api/getProduct/index.ts [hunk 7]
```diff
@@ -1,10 +1,36 @@
...
+  const defaultVariables = {
+    ...params
+  };
+
+  const { product } = context.extendQuery(
+    customQuery, { product: { query: defaultQuery, variables: defaultVariables } }
+  );
+
+  try {
+    const request = await (context.client as ApolloClient<any>).query<ProductData>({
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/getProduct/index.ts:22]: to achieve the above over here you need to destructor request.

File: packages/composables/src/getters/cartGetters.ts [hunk 8]
```diff
@@ -4,94 +4,97 @@ import {
...
+function getItemName(item: CartItem): string {
+  return 'Name';
+}
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const getCartItemImage = (product: any): string =>
-  product?.images[0] ||
-  "https://s3-eu-west-1.amazonaws.com/commercetools-maximilian/products/081223_1_large.jpg";
+function getItemImage(item: CartItem): string {
+  return 'https://s3-eu-west-1.amazonaws.com/commercetools-maximilian/products/081223_1_large.jpg';
```
Comment (Mateuszpietrusinski) [inline packages/composables/src/getters/cartGetters.ts:25]: add todo cuz now is hardcodeded

**Review by Mateuszpietrusinski — APPROVED** [review]
You need to do some iteration on smaller things, but even, for now, it's a huge improvement for this integration, great work.

**Pull request merged**